### PR TITLE
backlight-control-rs: init at 1.0.0

### PR DESCRIPTION
**Pull request by DOD-101**
Adding [backlight_control_rs](https://github.com/DOD-101/backlight_control_rs) as a package and a NixOS module. 

backlight_control_rs is a simple utility written in rust which allows the user to change the brightness of their backlight.

Originally I was going to package [backlight_control](https://github.com/Hendrikto/backlight_control/) for NixOS, but since this wasn't possible I went ahead and create an alternative with a few improvements and made it NixOS compatible. 

This is similar to [light](https://gitlab.com/dpeukert/light), but I still think, that this is a valuable alternative and considering that `backlight_control` was being used by at least a handful of people on Arch Linux, there are probably people using nix who would appreciate an analogous offering. 

I have also added myself as a maintainer, for both the package and the NixOS module.
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by DOD-101** [comment]
> Submit using the email address associated with GitHub

I fixed the e-mail issue. I had a typo in my config. 

Currently working on the rebasing.

File: pkgs/by-name/ba/backlight-control-rs/package.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  src = fetchFromGitHub {
+    owner = "DOD-101";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-yFQASeTmk1a9zmglE7s2Wg3EvNKeQ0qNjWvsCFLYX9U=";
+  };
+
+  cargoHash = "sha256-e6gZ8vYNzhqYbeYSS8x4YuTtaDDoZTBtuRtz32HHXS8=";
+
+  postInstall = ''
```
Comment (DOD-101) [inline pkgs/by-name/ba/backlight-control-rs/package.nix:20]: I had initially forgotten to add the postInstall phase for the udev rule.

**Review by ghost — APPROVED** [review]

File: pkgs/by-name/ba/backlight-control-rs/package.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  meta = {
+    description = "Backlight control is a re-written version of backlight_control with a few key improvements";
+    homepage = "https://github.com/DOD-101/backlight_control_rs";
+    mainProgram = "backlight_control_rs";
+    license = with lib.licenses; [
+      mit
+      asl20
+    ];
+    maintainers = with lib.maintainers; [ dod-101 ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ba/backlight-control-rs/package.nix:35]:
```suggestion
    maintainers = with lib.maintainers; [ dod-101 ];
    platforms = lib.platforms.linux;
```

As depending on Linux sysfs

File: pkgs/by-name/ba/backlight-control-rs/package.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    hash = "sha256-yFQASeTmk1a9zmglE7s2Wg3EvNKeQ0qNjWvsCFLYX9U=";
+  };
+
+  cargoHash = "sha256-e6gZ8vYNzhqYbeYSS8x4YuTtaDDoZTBtuRtz32HHXS8=";
+
+  postInstall = ''
+    substituteInPlace 90-backlight.rules --replace-fail /bin ${coreutils}/bin
+
+    mkdir -p $out/lib/udev/rules.d
+    ${coreutils}/bin/cp ./90-backlight.rules $out/lib/udev/rules.d
```
Comment (Aleksanaa) [inline pkgs/by-name/ba/backlight-control-rs/package.nix:24]:
```suggestion
    install -Dm644 ./90-backlight.rules -t $out/lib/udev/rules.d
```

File: pkgs/by-name/ba/backlight-control-rs/package.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  coreutils,
+  lib,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "backlight_control_rs";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "DOD-101";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/ba/backlight-control-rs/package.nix:13]:
```suggestion
    repo = "backlight_control_rs";
```

File: pkgs/by-name/ba/backlight-control-rs/package.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  postInstall = ''
+    substituteInPlace 90-backlight.rules --replace-fail /bin ${coreutils}/bin
+
+    mkdir -p $out/lib/udev/rules.d
+    ${coreutils}/bin/cp ./90-backlight.rules $out/lib/udev/rules.d
+  '';
+
+  meta = {
+    description = "Backlight control is a re-written version of backlight_control with a few key improvements";
```
Comment (Aleksanaa) [inline pkgs/by-name/ba/backlight-control-rs/package.nix:28]:
```suggestion
    description = "Re-written version of backlight_control with a few key improvements";
```

File: nixos/modules/programs/backlight-control-rs.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+in
+{
+  options = {
+    programs.backlight-control-rs = {
+      enable = lib.mkOption {
+        default = false;
+        type = lib.types.bool;
+        description = ''
+          Whether to install backlight-control-rs and udev rules granting access to members of the "video" group.
+        '';
```
Comment (Aleksanaa) [inline nixos/modules/programs/backlight-control-rs.nix:18]:
```suggestion
      enable = lib.mkEnableOption ''
        backlight-control-rs, and add udev rules granting access to members of the "video" group.
      '';
```

**Comment by Aleksanaa** [comment]
Not objections, but `systemd-logind` already provides a backlight control api via DBus, and [brightnessctl](https://search.nixos.org/packages?channel=unstable&show=brightnessctl&from=0&size=50&sort=relevance&type=packages&query=brightnessctl) consumes it, and in this way a udev file or setuid is not needed. Such program might be useful on "suckless" systems, but since NixOS bundles systemd, it is not particularly useful on NixOS.

File: pkgs/by-name/ba/backlight-control-rs/package.nix (hunk 4 above)
Comment (DOD-101) [reply pkgs/by-name/ba/backlight-control-rs/package.nix:13]: Why do this rather than using the pname variable?

**Comment by DOD-101** [comment]
Is there anything stopping this from getting merged?

**Pull request closed without merging**